src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public ShopServiceImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationHandlers.get(transaction.getOperation());
+            if (handler != null) {
+                handler.handle(transaction);
+            } else {
+                throw new RuntimeException("No handler found for operation: "

```

Comment:
Avoid using if-else to handle operation types. Consider applying the Strategy pattern more effectively by ensuring that the operationHandlers map is well-configured and all possible operations have corresponding handlers. This way, you can eliminate the need for the if-else check.